pkgs/development/libraries/libfprint-2-tod1-goodix/default.nix

```diff
@@ -25,6 +25,7 @@ stdenv.mkDerivation {
   '';
 
   passthru.driverPath = "/lib/libfprint-2/tod-1";
+  passthru.needsLibfprint_1_90 = true;  # https://gitlab.freedesktop.org/libfprint/libfprint/-/issues/484
```

Comment:
Are you certain that this is exhaustive? There are a few other Touch OEM Drivers. If this is not properly covering all the necessary ones, the next update of NixOS might break people's ability to enroll fingerprints or maybe even to login using fingerprint.

Reply:
I'm not at all sure this is exhaustive: I only added this for the one driver you gave me an issue link for. I don't have hardware for the other drivers, and so cannot test them (the linked issue details a runtime error). Do you think that pinning the version for all drivers unless proven to work would be a better idea? Perhaps just for the proprietary ones? Interested in your thoughts.

Reply:
I think the other way around would be safer, although I admit this is quite conservative. At the same time, I would not want to lock someone out of their system, because they rely on fingerprint auth and forgot their password or something like that, even though that's obviously on them.

Reply:
Alright, let's do that! :) I'll add the attribute for all drivers other than `vfs0090`. I've not tested that one, but I am using [a fork of it](https://gitlab.com/bingch/libfprint-tod-vfs0090) and have pulled in a patch from that which makes it compile on v1.92.0+, so it seems relatively likely that it'll work.

Besides that, do you have any idea what the upgrade path for this will look like? How do we best get users to test if the latest version works? I'd like to pave the way to us either having a bug report link attached to each of the pins, or the pin removed if determined to not be necessary.

Reply:
Pushed. Given that `vfs0900` is both the only driver that didn't have this pinned, and is the only non-proprietary driver that we package, I'm less concerned about collecting bug reports for the rest. I'm not going to hold my breath for them to receive updates. And I suppose maintainers (or adventurous users) of the individual drivers may come along and check— I think that the attribute being visible in the module's `defaultText` will go some way towards piquing their interest/raising awareness.

Reply:
How about guarding it by a [`stateVersion`](https://nixos.org/manual/nixos/stable/options#opt-system.stateVersion) check? I believe that is how the postgresql module handles possibly incompatible version upgrades.

Reply:
That would have to go purely in the module, so either a second attribute would have to be added, or the attribute would have to accept more states than just present/missing (e.g. present, missing, and "unknown"). That, plus the logic in the NixOS module would have to be made more complicated, to handle this additional condition. This isn't a stopper of course, but it's something to think about.